src/target/aby/trans.rs

```diff
@@ -0,0 +1,477 @@
+                            )),
+                        );
+                    }
+                    BvBinPred::Ult => {
+                        self.cache.insert(
+                            t.clone(),
+                            EmbeddedTerm::Bool(format!("circ->PutGTGate({}, {})", b, a)),
+                        );
+                    }
+                    _ => panic!("Non-field in bool BvBinPred: {}", op),
+                }
+            }
+            _ => panic!("Non-field in embed_bool: {}", t),
+        }
+
+        self.get_bool(&t)
+    }
+
+    /// Given term `t`, type-check `t` is of type Bv and return the variable name for
+    /// `t`
+    fn get_bv(&self, t: &Term) -> String {
+        match self
+            .cache
+            .get(t)
+            .unwrap_or_else(|| panic!("Missing wire for {:?}", t))
+        {
+            EmbeddedTerm::Bv(b) => b.clone(),
+            _ => panic!("Non-bv for {:?}", t),
+        }
+    }
+
+    fn embed_bv(&mut self, t: Term) -> String {
+        match &t.op {
+            Op::Var(name, Sort::BitVector(_)) => {
+                if !self.inputs.contains_key(name) {
+                    self.inputs
+                        .insert(name.to_string(), *self.md.inputs.get(name).unwrap());
+                    self.inputs_order.push(name.to_string());
+                }
+                if !self.cache.contains_key(&t) {
+                    self.cache
+                        .insert(t.clone(), EmbeddedTerm::Bv(format!("s_{}", name)));
+                }
+            }
+            Op::Const(Value::BitVector(b)) => {
+                self.cache.insert(
+                    t.clone(),
+                    EmbeddedTerm::Bool(format!(
+                        "circ->PutCONSGate((uint64_t){}, (uint32_t){})",
+                        b,
+                        b.width()
+                    )),
+                );
+            }
+            Op::Ite => {
+                let sel = self.get_bool(&t.cs[0]).clone();
+                let a = self.get_bv(&t.cs[1]).clone();
+                let b = self.get_bv(&t.cs[2]).clone();
+                self.cache.insert(
+                    t.clone(),
+                    EmbeddedTerm::Bool(format!("circ->PutMUXGate({}, {}, {})", a, b, sel)),
+                );
+            }
+            Op::BvNaryOp(o) => {
+                let a = self.get_bv(&t.cs[0]);
+                let b = self.get_bv(&t.cs[1]);
+
+                match o {
+                    BvNaryOp::Xor => {
+                        self.cache.insert(
+                            t.clone(),
+                            EmbeddedTerm::Bv(format!("circ->PutXORGate({}, {})", a, b)),
+                        );
+                    }
+                    BvNaryOp::Or => {
+                        self.cache.insert(
+                            t.clone(),
+                            EmbeddedTerm::Bv(format!(
+                                "((BooleanCircuit *)circ)->PutORGate({}, {})",
```

Comment:
I'm intrigued by this cast.

Reply:
Similar with the negation gate, OR apparently only works with Boolean Circuits... 
[Searching for PutORGate only shows boolean_circuits.cpp](https://github.com/encryptogroup/ABY/search?q=putorgate)

Reply:
Ah, makes sense. I thought this meant something else, which is why I was confused.